core/commit.go

```diff
@@ -154,15 +154,23 @@ func makeCommitmentCounter(f uint32) commitmentCounter {
 		if replicasCommitted[replicaID] {
 			return false, fmt.Errorf("Duplicated commitment")
 		}
-		replicasCommitted[replicaID] = true
 
-		if len(replicasCommitted) == int(f+1) {
-			delete(prepareStates, prepareCV)
-			lastDoneCV++
+		for cv := lastDoneCV + 1; cv < prepareCV; cv++ {
+			s := prepareStates[cv]
+			if len(s) != 0 && !s[replicaID] {
```

Comment:
Do we have to check `len(s) != 0` or simply non-nil check is enough?

Reply:
I think it has to be like this, because it gets assigned with an empty map at [line 151](https://github.com/hyperledger-labs/minbft/pull/123/files/9d6719f9ab366af1f1b970ecfd7d7ca049b83885#diff-7377bd6b40f03b94c9d00ac162e96100L151).

Reply:
However, the first commitment should be for a PREPARE from the primary. I need to give it a though... Thanks for noticing :wink: